src/forms/inputs/checkbox-group.js

```diff
@@ -109,7 +109,9 @@ function CheckboxGroup(props) {
     className,
     checkboxInputProps,
     ...rest
-  } = omitLabelProps(props)
+  } = props
+  const inputProps = omitLabelProps(rest)
```

Comment:
This is pretty wild. We got a _lot_ of `react/no-unused-prop-types` warnings when we upgraded to the last eslint v7 (even before v8). It looks like there was an issue with the implementation, because on `main` it doesn't work as you'd expect it to:
![](https://p-1MFmr4.b2.n0.cdn.getcloudapp.com/items/qGu6Bm8g/4b97dbb0-ebeb-4b9e-94f8-1cae06c8e865.jpg?source=client&v=72285ae7ac75fb83e9070c4b391c63e3)

`eslint` is not able to determine if the destructured props resulting from `omitLabelProps` are actually the props that you intended on using – fair enough. Just took a little bit of restructuring, which you'll see repeated across several components

Reply:
🤔 I would have expected `const inputProps = omitLabelProps(rest)` to pass in `props` instead of `rest` for `omitLabelProps`. Can you provide a bit more details on that change?

Reply:
Is `rest` here `props` without the `input, meta, options, className, checkboxInputProps` keys?

Reply:
If we were to store `omitLabelProps(props)` in the `inputProps` variable, then that variable would include `input, meta, options, className, checkboxInputProps`, which would then get passed _directly_ to the input (which we don't want to happen).

We don't _need_ to call `omitLabelProps(props)` when destructuring those keys because we aren't destructuring any of the label props in the first place!

To illustrate:
```js
const props = { label: 'First Name', input: { name: 'firstName', value: 'Conor', onChange: setFirstName }, 'aria-label': 'Employee First Name' }
const { input, ...rest } = props // rest -> { label: 'First Name', 'aria-label': 'Employee First Name' }
const inputProps = omitLabelProps(rest) // inputProps -> { 'aria-label': 'Employee First Name' }
```
Before, we were technically doing this all in one step, which confused the linter
```js
const props = { label: 'First Name', input: { name: 'firstName', value: 'Conor', onChange: setFirstName }, 'aria-label': 'Employee First Name' }
const { input, ...rest } = omitLabelProps(props) // rest -> { 'aria-label': 'Employee First Name' }
```

Reply:
Confused the linter and me 😉 

I follow now - I thought in addition to making the linter happy this might actually introduce a _slight_ difference in functionality in which `prop` keys are filtered but I understand the equivalence now. Thank you!